app/models/spree/gateway/flow_io.rb

```diff
@@ -56,18 +58,28 @@ def purchase(_amount, _payment_method, options = {})
         end
       end
 
-      def refund(_money, _authorization_key, options = {})
-        order = load_order options
-        order.cc_refund
+      def refund(payment, amount, _options = {})
+        order = payment.order
+        refund_form =
+          Io::Flow::V0::Models::RefundForm.new(order_number: order.number, amount: amount, currency: order.currency)
+        response = FlowcommerceSpree.client.refunds.post(FlowcommerceSpree::ORGANIZATION, refund_form)
+        response_status = response.status.value
+        if response_status == REFUND_SUCCESS
+          add_refund_to_order(response, order)
+          ActiveMerchant::Billing::Response.new(true, REFUND_SUCCESS, {}, {})
+        else
+          msg = "Partial refund fail. Details: #{response_status}"
```

Comment:
similar to below's questions. If this is going to be shown in the front end, I recommend storing the text in i18n files

Reply:
this is for Spree::Admin